server.js

```diff
@@ -1,22 +1,399 @@
+
+const authenticateUser = async (req, res, next) => {
+  const authHeader = req.headers.authorization;
+  if (!authHeader) {
+    return res.status(401).json({ message: 'No access token provided' });
+  }
+  // Accept either "Bearer <token>" or just "<token>"
+  const accessToken = authHeader.startsWith('Bearer ')
+    ? authHeader.replace('Bearer ', '')
+    : authHeader;
+
+  try {
+    const user = await User.findOne({ accessToken });
+    if (!user) {
+      return res
+        .status(401)
+        .json({ message: 'You need to login or sign up to post a thought' });
+    }
+    req.user = user;
+    next();
+  } catch (error) {
+    return res.status(401).json({ message: 'Invalid access token' });
+  }
+};
+
+const HappyThoughts = mongoose.model('HappyThoughts', happyThoughtsSchema);
+
+// Connect to MongoDB
+mongoose.connect(
+  process.env.MONGO_URL || 'mongodb://localhost/happythoughts'
+);
+mongoose.Promise = Promise;
+
+// RESET_DB logic
+if (process.env.RESET_DB) {
+  const seedDatabase = async () => {
+    console.log('Resetting database!');
+    await HappyThoughts.deleteMany();
+
+    for (const item of happythoughtsData) {
+      const newThought = new HappyThoughts(item);
+      await newThought.save();
+    }
+  };
+
+  seedDatabase();
+}
 
 // Add middlewares to enable cors and json body parsing
-app.use(cors())
-app.use(express.json())
+app.use(cors());
+app.use(express.json());
 
 // Start defining your routes here
-app.get("/", (req, res) => {
-  res.send("Hello Technigo!")
-})
 
-// Start the server
+//return all thoughts
+app.get('/thoughts', async (req, res) => {
+  try {
+    const thoughts = await HappyThoughts.find()
+      .sort({ createdAt: -1 })
+      .limit(20);
+    if (thoughts.length > 0) {
+      return res.json(thoughts);
+    } else {
+      res.status(404).json({ error: 'No thoughts available' });
+    }
+  } catch (error) {
+    res.status(400).json({ error: 'Invalid request' });
+  }
+});
+
+// return a random thought (has to be placed before the id route)
+app.get('/thoughts/random', async (req, res) => {
+  try {
+    const count = await HappyThoughts.countDocuments();
+    if (count === 0) {
+      return res.status(404).json({ error: 'No thoughts' });
+    }
+    const random = Math.floor(Math.random() * count);
+    const randomThought = await HappyThoughts.findOne().skip(random);
+
+    res.json(randomThought);
+  } catch (error) {
+    res.status(400).json({ error: 'Invalid request' });
+  }
+});
+
+// //return all thoughts sorted by likes (most likes on top)(has to be placed before the id route)
+// app.get('/thoughts/likes', async (req, res) => {
+//   try {
+//     const sortedThoughts = await HappyThoughts.find().sort({ hearts: -1 });
+//     if (sortedThoughts.length > 0) {
+//       res.json(sortedThoughts);
+//     } else {
+//       res.status(404).json({ error: 'No thoughts' });
+//     }
+//   } catch (error) {
+//     res.status(400).json({ error: 'Invalid request' });
+//   }
+// });
```

Comment:
If you dont use this, you could remove it. If you want to keep it to remember or fix later you can add that as a comment